### PR TITLE
Toponaming/part features chamfer, fillet; dependencies and test updates

### LINKED ISSUE #10357: [Issue] TopoNaming Phase 3 (card 5 of 35) - Add History Support/Part Workbench/FeatureChamfer
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Full version info

```shell
[code]
OS: Pop!_OS 22.04 LTS (pop:GNOME/pop)
Word size of FreeCAD: 64-bit
Version: 0.21.0.33675 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 0.21.0)
Hash: 41b058e2087de60dc8fef3d3e68c7d0129e13abf
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: English/United States (en_US)
[/code]
```

### Subproject(s) affected?

Part

### Problem description

This is a card for Toponaming fix Phase 3 (add naming) Part 1 (history expansion).

Please see TN issue https://github.com/FreeCAD/FreeCAD/issues/8432 for overall discussion.

Also see project https://github.com/orgs/FreeCAD/projects/2/views/1 for tracking.

Phase 3 does not USE the naming yet, that is Phase 4.

# Specifics

Naming is based on history. Add the RealThunder changes to history to expand it's support.

RealThunder reference branch: https://github.com/FreeCAD/FreeCAD/tree/development/toponaming

Do this for the workbench and file(s) mentioned in the issue title. The few cards are generic and the remaining ones are for specific `Feature` elements in the related workbench.

Please add one or more unit tests.

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #10358: [Issue] TopoNaming Phase 3 (card 6 of 35) - Add History Support/Part Workbench/FeatureFillet
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Full version info

```shell
[code]
OS: Pop!_OS 22.04 LTS (pop:GNOME/pop)
Word size of FreeCAD: 64-bit
Version: 0.21.0.33675 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 0.21.0)
Hash: 41b058e2087de60dc8fef3d3e68c7d0129e13abf
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: English/United States (en_US)
[/code]
```

### Subproject(s) affected?

Part

### Problem description

This is a card for Toponaming fix Phase 3 (add naming) Part 1 (history expansion).

Please see TN issue https://github.com/FreeCAD/FreeCAD/issues/8432 for overall discussion.

Also see project https://github.com/orgs/FreeCAD/projects/2/views/1 for tracking.

Phase 3 does not USE the naming yet, that is Phase 4.

# Specifics

Naming is based on history. Add the RealThunder changes to history to expand it's support.

RealThunder reference branch: https://github.com/FreeCAD/FreeCAD/tree/development/toponaming

Do this for the workbench and file(s) mentioned in the issue title. The few cards are generic and the remaining ones are for specific `Feature` elements in the related workbench.

Please add one or more unit tests.

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bgbsww**
Fixes #10358 
Fixes #10357

### CODE REVIEW

File: src/Mod/Part/App/PartFeature.h [hunk 1]
```diff
@@ -175,8 +175,15 @@ class FilletBase : public Part::Feature
 
     App::PropertyLink   Base;
     PropertyFilletEdges Edges;
+    App::PropertyLinkSub   EdgeLinks;
 
     short mustExecute() const override;
+    virtual void onUpdateElementReference(const App::Property *prop) override;
```
Comment (chennes) [inline src/Mod/Part/App/PartFeature.h:181]:
```suggestion
    void onUpdateElementReference(const App::Property *prop) override;
```

File: src/Mod/Part/App/PartFeature.h [hunk 2]
```diff
@@ -175,8 +175,15 @@ class FilletBase : public Part::Feature
 
     App::PropertyLink   Base;
     PropertyFilletEdges Edges;
+    App::PropertyLinkSub   EdgeLinks;
 
     short mustExecute() const override;
+    virtual void onUpdateElementReference(const App::Property *prop) override;
+
+protected:
+    virtual void onDocumentRestored() override;
```
Comment (chennes) [inline src/Mod/Part/App/PartFeature.h:184]:
```suggestion
    void onDocumentRestored() override;
```

File: src/Mod/Part/App/PartFeature.h [hunk 3]
```diff
@@ -175,8 +175,15 @@ class FilletBase : public Part::Feature
...
     App::PropertyLink   Base;
     PropertyFilletEdges Edges;
+    App::PropertyLinkSub   EdgeLinks;
 
     short mustExecute() const override;
+    virtual void onUpdateElementReference(const App::Property *prop) override;
+
+protected:
+    virtual void onDocumentRestored() override;
+    virtual void onChanged(const App::Property *) override;
```
Comment (chennes) [inline src/Mod/Part/App/PartFeature.h:185]:
```suggestion
    void onChanged(const App::Property *) override;
```

File: src/Mod/Part/App/TopoShapeExpansion.cpp [hunk 4]
```diff
@@ -831,6 +833,102 @@ void TopoShape::mapSubElement(const TopoShape& other, const char* op, bool force
...
+                        FC_WARN("hasher mismatch");
+                }else {
+                    // FC_THROWM(Base::RuntimeError, "hasher mismatch");
+                    FC_ERR("hasher mismatch");
+                }
+                Hasher = other.Hasher;
+            }
+        }else
+            Hasher = other.Hasher;
+    };
```
Comment (chennes) [inline src/Mod/Part/App/TopoShapeExpansion.cpp:864]: This code block looks familiar -- does it already exist someplace in Expansion?

Comment (bgbsww) [reply src/Mod/Part/App/TopoShapeExpansion.cpp:864]: It does, but a full search on getElementMapSize shows no other matches, except for some very similar code around line 823 - which is the other fork of the #else here.

Comment (chennes) [reply src/Mod/Part/App/TopoShapeExpansion.cpp]: OK, thanks for checking. I thing I stared at this myself for a while a month or so ago to try to get the complexity of this method down, so it's stuck in my brain.

**Pull request merged**